llpc/context/llpcPipelineContext.cpp

```diff
@@ -102,6 +102,10 @@ static cl::opt<unsigned> ShadowDescTablePtrHigh("shadow-desc-table-ptr-high",
 // -force-loop-unroll-count: Force to set the loop unroll count.
 static cl::opt<int> ForceLoopUnrollCount("force-loop-unroll-count", cl::desc("Force loop unroll count"), cl::init(0));
 
+// -disable-image-resource-check: disable image resource type check
```

Comment:
This should not be a command line option; it should be a pipeline option in LLPC like it is in LGC. Then the driver can decide whether to set it, perhaps on a per-app basis.

Reply:
I think driver can tune command line options per app. see https://github.com/GPUOpen-Drivers/xgl/blob/dev/icd/api/compiler_solution_llpc.cpp#L661 If I read correctly, pipeline option in llpc is used for tuning options per pipeline. I don't see the need for now. We just need per-app tuning for this newly add option.

Reply:
This is not a tuning option used frequently for different Apps. Instead, it is a workaround flag targeted at special Apps. So we can add it as a pipeline option.

Reply:
"disable-image-resource-check" is not a workaround flag targeted at special Apps. It is a performance tuning option to disable a workaround that may possibly hurt performance. With this patch, we will by-default check against invalid buffer descriptor when accessing image descriptors for GFX10. This may hurt some app's performance. If we found more apps been hurt by this workaround, we need to enable this option for that app. So it is hard to say this is not-frequently used option.

Reply:
Well, I guess in most cases, we ought to disable the workaround just like we don't have such workaround now. And for some particular Apps, we will conditionally enable the workaround. So if we have a flag in the pipeline option, it is something like this:

{
    bool enableImageResourceCheck;
}

In most cases, we don't set the flag and it is false by default. For those Apps that need such check, set it to true. I don't have any preference. A command line option is fine to me as well but LGC would like to restrict the increase of command line options since they are not exposed clearly as interfaces for use.

Reply:
well the driver part code is ready, see https://github.com/GPUOpen-Drivers/xgl/blob/dev/icd/settings/settings.cpp#L476. I think quality is top-priority behind the decision, which means we want to make sure "even unknown apps do wrong, the hardware will not crash". We don't know if there are other apps doing wrong, and seems we don't want to be annoyed by such kind of things again. I will switch to use pipeline option if others also agree it is the best place.